MovingBlocks/Terasology
# test(MTE): replace IsolatedMTEExtension with JUnit's TestInstance.Lifecycle

This removes the `@IsolatedMTEExtension` variety of MTE test in favor of using JUnit's TestInstance.Lifecycle to manage sharing things between test methods, as proposed in Terasology/ModuleTestingEnvironment#75.

This simplifies things and fixes #5038.

The first commit here https://github.com/MovingBlocks/Terasology/pull/5039/commits/a43627f461fc3a1fe47d58e826511765b8d91c6c contains a test case that reproduced the failure described in #5038.

In practice, this is mostly compatible with existing MTE tests, but they may need some adjustments. For specifics, see https://github.com/MovingBlocks/Terasology/pull/5039#issuecomment-1145437325

## Related
- [x] #5037 would be nice to have, as it would let us separate this bug from the "port already in use" failure of trying to start two _servers_ on the same port.
- [x] #5022 would be nice to have, as both PRs edit MTEExtension and conflicts are unavoidable.

### How to test

Run MTE tests of modules. Note that the modules must import MTE from `org.terasology.integrationtest` to use this code; check for module PRs linked to #5010.
